src/Database/Beam/Backend.hs

```diff
@@ -116,11 +116,17 @@ openDatabase' isDebug db strat dbSettings =
 
      return beam'
 
+
 dumpSchema :: Database db => DatabaseSettings db -> IO ()
-dumpSchema (db :: DatabaseSettings db) =
+dumpSchema (db :: DatabaseSettings db) = do
+  putStrLn "Dumping database schema ..."
+  schema <- showSchema db
+  mapM_ putStrLn $ lines schema
+
+showSchema :: Database db => DatabaseSettings db -> IO String
```

Comment:
As dumpSchema returned IO (), I couldn't capture the result, which I needed for tests.